src/components/OTP.vue

```diff
@@ -0,0 +1,337 @@
+    },
+
+    /** Checks if any phoneNumber has been entered */
+    isAnyphoneNumberPresent() {
+      return this.phoneNumberList != undefined && this.phoneNumber != "";
+    },
+
+    /** Whether the 'Request OTP' button should be disabled */
+    isRequestOTPButtonDisabled() {
+      return (
+        this.phoneNumber == undefined ||
+        this.invalidInputMessage != "" ||
+        this.isOTPSent == true
+      );
+    },
+
+    /**
+     * Whether the submit button is disabled
+     * Returns true if any of the following conditions are met:
+     * - if 'Request OTP' button is enabled
+     * - if OTP hasn't been typed yet
+     */
+    isSubmitButtonDisabled() {
+      return !(this.isRequestOTPButtonDisabled || this.OTPCode.length > 0);
+    },
+
+    /** Checks if the current input entry has the required number of characters */
+    isCurrentEntryIncomplete() {
+      return this.phoneNumber.length < this.maxLengthOfId;
+    },
+
+    /** Whether input being typed is in the correct format */
+    isInvalidInputMessageShown() {
+      return this.invalidInputMessage != null;
+    },
+
+    /** Returns the heading text for the input box */
+    inputBoxDisplayText() {
+      return this.programData.text.default.display;
+    },
+
+    /** Returns the text for the OTP code input box */
+    enterOTPInputBoxDisplayText() {
+      return this.programData.text.default.enterOTP;
+    },
+
+    /** Returns the text for the button to request an OTP */
+    requestOTPButtonDisplayText() {
+      return this.programData.text.default.requestOTP;
+    },
+
+    /** Returns the text for the submit button */
+    submitButtonDisplayText() {
+      return this.programData.text.default.submitButton;
+    },
+
+    /** Returns the maximum length of the ID */
+    maxLengthOfId() {
+      return this.programData.input.maxLengthOfId;
+    },
+
+    /** Returns the basic validation type for the input */
+    basicValidationType() {
+      return this.programData.input.basicValidationType;
+    },
+
+    /** Returns the invalid input message stored against each program */
+    invalidInputText() {
+      return this.programData.text.default.invalid.input;
+    },
+  },
+  methods: {
+    /** Determines how the input box should look.
+     * - If an input error needs to be displayed, the box has a a red border.
+     * - Otherwise, it has an opacity of 30.
+     */
+    selectInputBoxClasses() {
+      let baseStyle =
+        "border-2 rounded-sm p-4 mx-auto border-gray-500 focus:border-gray-800 focus:outline-none";
+      return this.invalidInputMessage
+        ? baseStyle + "border-red-600 focus:border-red-600"
+        : baseStyle + "pointer-events-none opacity-30";
+    },
+
+    /** Calls the mapping function to validate the typed character
+     * @param {Object} event - event triggered when a character is typed
+     */
+    isValidPhoneNumber(event) {
+      if (validationTypeToFunctionMap[this.basicValidationType](event)) {
```

Comment:
Since this is the OTP component, do we need this `validationTypeToFunctionMap`? Don't we know that it will be a phone number?

Reply:
Yeah, since it's OTP, the `basicValidationType` will be `numeric`. Just wanted it to be consistent across components.

Reply:
If consistent across components and same/similar validation methods are being used, why not combine the common code into one place?

Reply:
The validation methods are in one place. I was talking about syntax consistency. As in, instead of hardcoding the value, referring to the mapping.